From e23c976b9c2316efcd9fc3bd3ec33c3872bb375f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Wed, 21 Dec 2022 17:26:11 +0100
Subject: [PATCH 262/954] feat(stm32mp2): add trace for SoC revision B

Add STM32MP2_REV_B define and its management in stm32mp_get_soc_name()
to have "Rev.B" displayed on this SoC version.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git eaf2e62b6faad3ecd836d8affbdc79abbdbb7abd feat(stm32mp2): add trace for SoC revision B]
Signed-off-by: Yann Gautier <yann.gautier@st.com>
Change-Id: I198739abc32579c0e6f1623a45f31efeeb6ac821
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/282349
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     | 1 +
 plat/st/stm32mp2/stm32mp2_private.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 8bddb2ed15..3feeebe705 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -52,6 +52,7 @@
 #define STM32MP257F_PART_NB	U(0x80002000)
 
 #define STM32MP2_REV_A		U(0x1000)
+#define STM32MP2_REV_B		U(0x2000)
 
 /*******************************************************************************
  * PACKAGE ID
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index 5f3f6dce7d..ec595e517e 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -290,6 +290,9 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 	case STM32MP2_REV_A:
 		cpu_r = "A";
 		break;
+	case STM32MP2_REV_B:
+		cpu_r = "B";
+		break;
 	default:
 		cpu_r = "?";
 		break;
-- 
2.53.0

